### PR TITLE
LPS-149034 [CP 2.0] Create logic to update status and success alert

### PR DESCRIPTION
**Pull request by Lucas2691**
### **How it was developed?**

Created logic to confirm `projectID` with input modal field, update `subscriptionsGroup` status after submit and show confirmation message.

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js [hunk 1]
```diff
@@ -182,11 +182,28 @@ export const addDXPCloudEnvironment = gql`
 	}
 `;
 
+export const updateDXPCloudEnvironment = gql`
+	mutation putDXPCloudProjectId(
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:186]:
```suggestion
	mutation updateDXPCloudProjectId(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/DXPCloud/index.js [hunk 2]
```diff
@@ -21,9 +22,11 @@ import {
 	getAccountSubscriptionsTerms,
 } from '../../../../../common/services/liferay/graphql/queries';
 import getActivationStatusDateRange from '../../../../../common/utils/getActivationStatusDateRange';
+import {ALERT_UPDATE_DXP_CLOUD_STATUS} from '../../../containers/DXPActivationKeysTable/utils/constants/alertUpdateDXPCloudStatus';
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/DXPCloud/index.js:25]: This utils should be exported from utils/constants and not being imported directly. Import them at the root of constants and import this root here
```suggestion
import {ALERT_UPDATE_DXP_CLOUD_STATUS} from '../../../containers/DXPActivationKeysTable/utils/constants';
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js [hunk 3]
```diff
@@ -9,11 +9,120 @@
...
+	const handleOnConfirm = () => {
+		const errorMessageGroupId = isLowercaseAndNumbers(projectID);
+
+		if (errorMessageGroupId) {
+			setHasError(errorMessageGroupId);
+
+			return;
+		}
+		updateSubscriptionGroupsStatus(accountKey);
+		updateProjectId(accountKey);
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js:54]: Setting both states with same value?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js [hunk 4]
```diff
@@ -9,11 +9,120 @@
...
+			await client.mutate({
+				mutation: updateAccountSubscriptionGroups,
+				variables: {
+					accountSubscriptionGroup: {
+						activationStatus: STATUS_TAG_TYPE_NAMES.active,
+					},
+					id: dxpCloudSubscriptionGroups?.accountSubscriptionGroupId,
+				},
+			}),
+		]);
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js:79]:
```suggestion
		
			await client.mutate({
				mutation: updateAccountSubscriptionGroups,
				variables: {
					accountSubscriptionGroup: {
						activationStatus: STATUS_TAG_TYPE_NAMES.active,
					},
					id: dxpCloudSubscriptionGroups?.accountSubscriptionGroupId,
				},
			}),
		
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss [hunk 5]
```diff
@@ -41,11 +41,15 @@
 .cp-activation-key-icon {
 	border: 1px solid var(--color-neutral-2);
 	box-shadow: none;
-	left: 641px;
+	left: 648px !important;
 	min-width: fit-content;
-	top: 276px;
-	ul li span {
+	top: 276px !important;
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss:46]: This is weird, why are you using static positioning here? I think that a better solution is to provide the parent element as a **position: relative** following the cp-activation-key-icon as a **position: absolute** and you can use the top and left props easily.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss [hunk 6]
```diff
@@ -41,11 +41,15 @@
...
+	ul li button {
 		padding: 0.75rem;
 		padding-right: 68px !important;
 	}
 }
+
+.cp-activation-status-container .card-body {
+	padding-right: 13px !important;
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss:54]: Do not use paddings/margins in CSS customization, our pattern is to always use even numbers. So I think is a better solution to use className="pr-3"(16px) OR pr-2 (8px) + pr-1 (4px) (two different elements)

**Review by iagochaves — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js (hunk 1 above)
Comment (Lucas2691) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:186]: Yes, @iagochaves

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/DXPCloud/index.js (hunk 2 above)
Comment (Lucas2691) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/DXPCloud/index.js:25]: Ok, man. Thks!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss (hunk 6 above)
Comment (Lucas2691) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss:54]: Thanks @iagochaves . Done!

**Review by joanagrego — APPROVED** [review]
Nice!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js [hunk 7]
```diff
@@ -9,11 +9,120 @@
...
+	onClose,
+	projectID,
+	projectIdValue,
+	setHasFinishedUpdate,
+	setProjectIdValue,
+	setSubscriptionGroupActivationStatus,
+}) => {
+	const [hasError, setHasError] = useState();
+
+	const [, dispatch] = useCustomerPortal();
```
Comment (joaosvassis) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js:43]:
```suggestion
	const [{subscriptionGroups}, dispatch] = useCustomerPortal();
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js [hunk 8]
```diff
@@ -9,11 +9,120 @@
...
+	const updateSubscriptionGroupsStatus = async (accountKey) => {
+		const {data: dataSubscriptionGroups} = await client.query({
+			query: getAccountSubscriptionGroups,
+			variables: {
+				filter: `accountKey eq '${accountKey}' and hasActivation eq true`,
+			},
+		});
+
+		const dxpCloudSubscriptionGroups =
+			dataSubscriptionGroups.c?.accountSubscriptionGroups?.items[0];
```
Comment (joaosvassis) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js:67]:
```suggestion
		const dxpCloudSubscriptionGroups =
			subscriptionGroups;
```

**Review by joaosvassis — APPROVED** [review]
Nice!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/DXPCloud/index.js (hunk 2 above)
Comment (Lucas2691) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/DXPCloud/index.js:25]: OK!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js (hunk 8 above)
Comment (Lucas2691) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ModalDXPCActivationStatus/index.js:67]: Thanks @joaosvassis ! Done.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss (hunk 5 above)
Comment (Lucas2691) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss]: Nice!

**Review by iagochaves — CHANGES_REQUESTED** [review]

**Comment by iagochaves** [comment]
ci:forward

**Review by jairmedeiros — APPROVED** [review]

**Review by igordmf — APPROVED** [review]
Good job

**Comment by jairmedeiros** [comment]
ci:forward

**Pull request closed without merging**